glean-core/src/event_database/mod.rs

```diff
@@ -189,34 +236,39 @@ impl EventDatabase {
             return;
         }
 
-        // Create RecordedEvent object, and its JSON form for serialization
-        // on disk.
-        let event = RecordedEvent {
-            timestamp,
-            category: meta.category.to_string(),
-            name: meta.name.to_string(),
-            extra,
-        };
-        let event_json = serde_json::to_string(&event).unwrap(); // safe unwrap, event can always be serialized
-
-        // Store the event in memory and on disk to each of the stores.
-        let mut stores_to_submit: Vec<&str> = Vec::new();
+        let mut submit_max_capacity_event_ping = false;
         {
             let mut db = self.event_stores.write().unwrap(); // safe unwrap, only error case is poisoning
             for store_name in meta.send_in_pings.iter() {
                 let store = db.entry(store_name.to_string()).or_insert_with(Vec::new);
-                store.push(event.clone());
+                let execution_counter = CounterMetric::new(CommonMetricData {
+                    name: "execution_counter".into(),
+                    category: store_name.into(),
+                    send_in_pings: vec![INTERNAL_STORAGE.into()],
+                    lifetime: Lifetime::Ping,
+                    ..Default::default()
+                })
```

Comment:
The `execution_counter` is never actually set/incremented anywhere. Therefore this will always be `None`.
That means:

* We're missing an `execution_counter.add_sync(&glean, 1)` somewhere
* We don't have a test that captures this behavior properly (might be easiest to "reset" Glean in Kotlin to try out this behavior

I added an event to the RLB prototype example (`glean-core/rlb/examples/prototype.rs`), removed the ping submission there, and then ran it several times.